fits2caom2/src/main/java/ca/nrc/cadc/caom2/fits/wcs/Wcs.java

```diff
@@ -379,57 +379,110 @@ public static CoordFunction2D getCoordFunction2D(String utype, FitsMapping mappi
+
+            String kw11 = String.format("PC%d_%d", posAxis1, posAxis1);
+            String s11 = mapping.getKeywordValue(kw11);
+            if (s11 == null) {
+                kw11 = String.format("PC0%d_0%d", posAxis1, posAxis1);
+                s11 = mapping.getKeywordValue(kw11);
+            }
+            if (s11 != null) {
+                pc11 = Double.parseDouble(s11);
+            }
+
+            String kw12 = String.format("PC%d_%d", posAxis1, posAxis2);
+            String s12 = mapping.getKeywordValue(kw12);
+            if (s12 == null) {
+                kw12 = String.format("PC0%d_0%d", posAxis1, posAxis2);
+                s12 = mapping.getKeywordValue(kw12);
+            }
+            if (s12 != null) {
+                pc12 = Double.parseDouble(s12);
+            }
+
+            String kw21 = String.format("PC%d_%d", posAxis2, posAxis1);
+            String s21 = mapping.getKeywordValue(kw21);
+            if (s21 == null) {
+                kw21 = String.format("PC0%d_0%d", posAxis2, posAxis1);
+                s21 = mapping.getKeywordValue(kw21);
+            }
+            if (s21 != null) {
+                pc21 = Double.parseDouble(s21);
+            }
+
+            String kw22 = String.format("PC%d_%d", posAxis2, posAxis2);
+            String s22 = mapping.getKeywordValue(kw22);
+            if (s22 == null) {
+                kw22 = String.format("PC0%d_0%d", posAxis2, posAxis2);
+                s22 = mapping.getKeywordValue(kw22);
+            }
+            if (s22 != null) {
+                pc22 = Double.parseDouble(s22);
+            }
+
+            if (pc11 != null && pc22 != null && pc12 == null && pc21 == null) {
+                pc12 = 0.0;
+                pc21 = 0.0;
+            }
+
+            if (pc11 != null && pc12 != null && pc21 != null && pc22 != null) {
+                double cd1 = Double.parseDouble(cdelt1);
+                double cd2 = Double.parseDouble(cdelt2);
+                cd11 = pc11 * cd1;
+                cd12 = pc12 * -cd2;
+                cd21 = pc21 * cd1;
+                cd22 = pc22 * cd2;
+
+                return new CoordFunction2D(dimension, refCoord, cd11, cd12, cd21, cd22);
             }
         }
-        
-        // If any of the cd matrix are null, return null.
-        if (cd11 == null || cd12 == null || cd21 == null || cd22 == null)
-            return null;
-        
+
+        // Try CROTA with PC matrix or default PC matrix
+        String crota = mapping.getKeywordValue("CROTA" + posAxis1);
+        if (crota == null) {
+            crota = mapping.getKeywordValue("CROTA" + posAxis2);
+        }
+        if (crota == null) {
+            crota = mapping.getKeywordValue("CROTA");
+        }
+        String crotsign = mapping.getKeywordValue("CROTSIGN");
+        double rotation = (crota == null ? 0.0 : Double.parseDouble(crota));
+        double sign = (crotsign == null ? 1.0 : Double.parseDouble(crotsign));
+
+        // Use the default PC matrix
```

Comment:
This is not consistent with the old code (line ~420) and even that seems to apply the CDELT values inconsistently with the above comment about CD = S*PC.... needs to be verified (how: TBD)